### PR TITLE
ARROW-3574: [Plasma] Use static libraries in plasma library.

### PR DESCRIPTION
**Pull request by robertnishihara**
I missed one location in https://github.com/apache/arrow/pull/2792. That previous PR fixed the issue on Linux but not on Mac. Without this PR, moving the `plasma_store_server` executable and then executing it leads to

```
dyld: Library not loaded: @rpath/libarrow.12.dylib
  Referenced from: /Users/rkn/Workspace/ray/./python/ray/core/src/plasma/plasma_store_server
  Reason: image not found
Abort trap: 6
```

### CODE REVIEW

File: cpp/src/plasma/CMakeLists.txt [hunk 1]
```diff
@@ -94,8 +94,8 @@ ADD_ARROW_LIB(plasma
   SOURCES ${PLASMA_SRCS}
   OUTPUTS PLASMA_LIBRARIES
   DEPENDENCIES gen_plasma_fbs
-  SHARED_LINK_LIBS ${FLATBUFFERS_STATIC_LIB} ${CMAKE_THREAD_LIBS_INIT} ${PLASMA_LINK_LIBS}
-  STATIC_LINK_LIBS ${FLATBUFFERS_STATIC_LIB} ${CMAKE_THREAD_LIBS_INIT} ${PLASMA_LINK_LIBS})
+  SHARED_LINK_LIBS ${FLATBUFFERS_STATIC_LIB} ${CMAKE_THREAD_LIBS_INIT} ${PLASMA_STATIC_LINK_LIBS}
```
[the remark this answers is not in the document]
Comment (robertnishihara) [reply cpp/src/plasma/CMakeLists.txt]: If we take that approach, is there any way to get `libarrow.so` to statically link into the `plasma_store_server` executable? Currently it is dynamically linked. cc @pcmoritz

[the remark this answers is not in the document]
Comment (wesm) [reply cpp/src/plasma/CMakeLists.txt]: You have to link plasma_static and arrow_static into the executable. You should not link libarrow.so with libplasma.a

**Review by pcmoritz — APPROVED** [review]
+1

**Comment by pitrou** [comment]
This seems to have broken CUDA builds: see [ARROW-3598](https://issues.apache.org/jira/browse/ARROW-3598).

**Comment by robertnishihara** [comment]
@pitrou, I created https://github.com/apache/arrow/pull/2820.

**Pull request closed without merging**